azurelinuxagent/common/protocol/util.py

```diff
@@ -144,50 +160,75 @@ def _set_wireserver_endpoint(self, endpoint):
             raise OSUtilError(ustr(e))
    
     def _detect_wire_protocol(self):
-        endpoint = self.dhcp_handler.endpoint
-        if endpoint is None:
-            logger.info("WireServer endpoint is not found. Rerun dhcp handler")
-            try:
-                self.dhcp_handler.run()
-            except DhcpError as e:
-                raise ProtocolError(ustr(e))
+        """
+        detect the wire protocol.
+        """
+        self.wire_lock.acquire()
+        try:
+            if self.wire_protocol != None:
+                return self.wire_protocol
```

Comment:
Use `if x is not None`; do not use `if x != None`. PEP-8 compliance, proper use of python, etc.